src/simulations/collect.jl

```diff
@@ -68,10 +68,13 @@ By default both keywords are `nothing`, i.e. nothing is collected/aggregated.
 """
 function run! end
 
-run!(model::ABM, agent_step!, n::Int = 1; kwargs...) =
+run!(model::ABM, agent_step!::Function, n::Int = 1; kwargs...) =
 run!(model::ABM, agent_step!, dummystep, n; kwargs...)
 
-function run!(model::ABM, agent_step!, model_step!, n;
+run!(model::ABM, agent_step!::Function, n::Function; kwargs...) =
+run!(model::ABM, agent_step!, dummystep, n; kwargs...)
```

Comment:
This dissallows `agent_step!` to be a function like object, and is a big limitation. https://docs.julialang.org/en/v1.5/manual/methods/#Function-like-objects

Reply:
Ah, we can probably drop that function guard.